### PR TITLE
Set map iterators

### PR DESCRIPTION
**Pull request by VitoCastellana**

### CODE REVIEW

File: include/shad/data_structures/compare_and_hash_utils.h [hunk 1]
```diff
@@ -171,6 +171,10 @@ uint64_t HashFunction(const std::vector<KeyTy> &key, uint8_t seed) {
   return hash;
 }
 
+template <>
+uint64_t HashFunction(const size_t &key, uint8_t) {
```
Comment (droccom) [inline include/shad/data_structures/compare_and_hash_utils.h]: mark const

File: include/shad/data_structures/hashmap.h [hunk 2]
```diff
@@ -617,6 +660,169 @@ void Hashmap<KTYPE, VTYPE, KEY_COMPARE, INSERT_POLICY>::AsyncApply(
...
+  }
+  
+  bool operator== (const map_iterator& other) const {
+    return (data_ == other.data_);
+  }
+  bool operator!= (const map_iterator& other) const {
+    return !(*this == other);
+  }
+
+  T operator*() {
```
Comment (droccom) [inline include/shad/data_structures/hashmap.h]: mark const

File: include/shad/data_structures/local_hashmap.h [hunk 3]
```diff
@@ -1014,6 +1057,93 @@ void LocalHashmap<KTYPE, VTYPE, KEY_COMPARE, INSERTER>::AsyncInsert(
...
+    return lmap_iterator(nullptr, numBuckets, 0, nullptr, nullptr);
+  }
+  bool operator== (const lmap_iterator& other) const {
+    return entryPtr_ == other.entryPtr_;
+  }
+  bool operator!= (const lmap_iterator& other) const {
+    return !(*this == other);
+  }
+
+  T operator*() {
```
Comment (droccom) [inline include/shad/data_structures/local_hashmap.h]: mark const

File: include/shad/data_structures/local_set.h [hunk 4]
```diff
@@ -191,6 +230,13 @@ class LocalSet {
       }
       return entries.get()[i];
     }
+    
+    Entry* getNextPtr(size_t& pos, Bucket*& bucketPtr) {
+      ++pos;
+      if (pos < bucketSize_) {
```
Comment (droccom) [inline include/shad/data_structures/local_set.h]: what is the point of this check?

File: include/shad/data_structures/local_set.h [hunk 5]
```diff
@@ -609,6 +658,93 @@ void LocalSet<T, ELEM_COMPARE>::AsyncForEachElement(rt::Handle& handle,
...
+    return lset_iterator(nullptr, numBuckets, 0, nullptr, nullptr);
+  }
+  bool operator== (const lset_iterator& other) const {
+    return entryPtr_ == other.entryPtr_;
+  }
+  bool operator!= (const lset_iterator& other) const {
+    return !(*this == other);
+  }
+
+  T operator*() {
```
Comment (droccom) [inline include/shad/data_structures/local_set.h]: mark const

**Comment by mminutoli** [comment]
Can we close this issue?

**Pull request merged**